pkg/domain/snapshots/snapshot.go

```diff
@@ -24,21 +24,18 @@ import (
 	"github.com/google/uuid"
 )
 
-// Snapshotter implements basic snapshot creation for audit formatters
-type Snapshotter[T es.Projection] struct {
+type Snapshot[T es.Projection] struct {
```

Comment:
I'm still debating my solution to the NP problem of naming hier so please let me know if snapshooter still makes more sense or something else.

Reply:
Hm, I think your name "Snapshotter" is fine. Since a snapshot itself is something different than what this struct represents.